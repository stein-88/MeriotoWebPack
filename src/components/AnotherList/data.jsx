const schemasnotherlist = [
    {
        id: '04b3aa0f-fbab-45ca-84fb-47e942a33b01',
        pubDate: '2022-04-30T11:00:00+00:00',
        url: null,
        newsType: 'howto',
        title: 'How to request an item.',
        subtitle: null,
        attributes: null,
        image: null,
        video: {
            url: 'video/b49c0163-dfc1-421c-8a01-139a336cdbd0.mp4',
        },
    },
    {
        id: '548947d2-3eda-4cfa-aa14-51a6ed4b4623',
        pubDate: '2022-04-18T17:19:14.155803+00:00',
        url: 'firstOne',
        newsType: 'feature',
        title: 'Noice Feature edited',
        subtitle: 'Check it out this new feature with sub edited',
        attributes: null,
        image: {
            url: 'image/e908f02c-c842-4aa7-a5d6-c363da806dfb.png',
        },
        video: null,
    },
    {
        id: 'c5986236-5c82-4736-a523-12a0f9565b43',
        pubDate: '2022-04-18T17:18:41.818124+00:00',
        url: 'https://developer.globechain.com/',
        newsType: 'news',
        title: 'Noice website',
        subtitle: 'Nice things that we will going to see in this API eidted',
        attributes: null,
        image: {
            url: 'image/ea97ff85-9805-44c1-8ecc-964a7125a619.png',
        },
        video: null,
    },
    {
        id: '1d12f4f9-3609-44ff-834f-b2cc630fdd09',
        pubDate: '2022-04-20T12:49:47.528002+00:00',
        url: null,
        newsType: 'news',
        title: 'Globechain is worldwide.',
        subtitle: 'asda',
        attributes: null,
        image: null,
        video: null,
    },
    {
        id: '6402c496-6e5d-486d-a2d6-6d4417c06c48',
        pubDate: '2022-04-20T11:11:38.769865+00:00',
        url: 'table',
        newsType: 'feature',
        title: 'Dashboard Filter',
        subtitle: 'asda',
        attributes: null,
        image: {
            url: 'image/17ca40d4-8d36-4dd4-b6b4-7f4b24c0a31a.png',
        },
        video: null,
    },
    {
        id: '881fb0b8-7295-4759-bc7f-25cb77a92ceb',
        pubDate: '2022-04-19T15:49:30.859416+00:00',
        url: 'logistic',
        newsType: 'feature',
        title: 'Noice logistic',
        subtitle: 'New feature',
        attributes: null,
        image: {
            url: 'image/49c1931b-46c2-49ba-aaf7-3a89dfc7ad9d.png',
        },
        video: null,
    },
    {
        id: '9da715cf-2d72-4979-99c1-d2b9dfe7d85e',
        pubDate: '2022-05-01T11:00:00+00:00',
        url: null,
        newsType: 'howto',
        title: 'Video How to do something',
        subtitle: null,
        attributes: '[{"id": "18ba7b6-6fd1-0547-52eb-47f8bf7ee1ea", "title": "Look how cool", "subtitle": "Noice ", "timestamp": 1650302383942}, {"id": "3fee1ce-ecab-b820-c4c4-e475ae56828", "title": "Very noice title editing", "imgUrl": "image/4b0d0b7f-d1bb-40b4-9d49-18bdda2f0aa9.png", "subtitle": "Withsub and image", "timestamp": 1650302400154}, {"id": "b11a48e-4d7a-dc1f-2b34-40c4ccffe6a", "subtitle": "Just sub now", "timestamp": 1650302406293}, {"id": "7ebcc18-c6ca-7cd6-14a4-b1dad105abd", "subtitle": "Followed from another sub.", "timestamp": 1650302412506}]',
        image: null,
        video: {
            url: 'video/53cf5355-162d-4dc2-a613-37331fb2e8b9.mp4',
        },
    },
]

export default schemasnotherlist
