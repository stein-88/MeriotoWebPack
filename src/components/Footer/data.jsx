const schema = {
    text: 'Â© 2022 Globechain. All Rights Reserved.',
}

export default schema
