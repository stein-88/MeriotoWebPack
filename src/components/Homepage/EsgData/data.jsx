const schema = {
    titulo: 'Globechain the ESG data reuse solution making waste a resource for everyone',
    text: [
        'We are the leading ESG data reuse marketplace connecting enterprises with non profits, small businesses and people to redistribute unneeded items.',
        'We generate ESG data on your impact, help you reduce waste costs and achieve your sustainability and SDG targets.',
    ],
}

export default schema
