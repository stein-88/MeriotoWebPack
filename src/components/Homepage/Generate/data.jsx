const schema = {
    titulo: 'Generate Impact',
    text: [
        'Our marketplace allows you to effortlessly manage and track your items, reduce costs, increase transparency and generate ESG data for your reporting.',
        'We offer external reuse via our marketplace, internal reuse and loaning asset inventory system.',
    ],
}

export default schema
