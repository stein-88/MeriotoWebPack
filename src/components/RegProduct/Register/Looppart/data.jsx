const schemalooppart = {
    delet: 'Delete Image',
    upimg: 'Update Image',
    inp: [
        {
            type: 'text',
            place: 'Product Name',
            refer: 'theproduct',
        },
        {
            type: 'number',
            place: 'Product Price',
            refer: 'theprice',
        },
        {
            type: 'number',
            place: 'Product Amount',
            refer: 'theamount',
        },
    ],
    fileinp: {
        type: 'file',
        place: 'Set the image',
        accept: 'image/*',
        refer: 'fileimg',
    },
}

export default schemalooppart
