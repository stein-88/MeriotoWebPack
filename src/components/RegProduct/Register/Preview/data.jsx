const schemabotoesPreview = {
    check: 'Check the list',
    sub: 'Submit',
    uni: 'units in stock',
    reset: 'Reset',
}

export default schemabotoesPreview
