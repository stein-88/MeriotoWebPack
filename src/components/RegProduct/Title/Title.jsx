import React from 'react'
import schema from './data'

const Title = ({ idserver }) => {
    if (!schema) return null
    const { titleedi, titlereg, subconst } = schema
    return (
        <section className="w-100 py-5 text-center">
            <div className="container">
                <div className="row">
                    {titleedi && titlereg
                        && (
<div className="col-6">
                            <section className="w-100">
                                <p
                                    data-test={!idserver ? titlereg : `${titleedi} - ${idserver}`}
                                    className="h1"
                                >
{!idserver ? titlereg : `${titleedi} - ${idserver}`}
                                </p>
                            </section>
</div>
)}
                    {subconst
                        && (
<div className="col-6">
                            <section className="w-100">
                                <p className="h1">{subconst}</p>
                            </section>
</div>
)}
                </div>
            </div>
        </section>
    )
}

export default Title
