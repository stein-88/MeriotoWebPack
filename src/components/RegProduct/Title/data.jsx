const schematitleprod = {
    titlereg: 'Register the Product',
    titleedi: 'Edit the Product',
    subconst: 'See how will be',
}

export default schematitleprod
