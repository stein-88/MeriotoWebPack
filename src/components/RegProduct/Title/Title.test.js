import React from 'react'
import renderer from 'react-test-renderer'
import Title from './Title'
import * as schemaMocked from './data'

describe('Test on Title Component', () => {
    test('Should render', () => {
        const compTitle = renderer.create(<Title />)
        const resultRoot = compTitle.root.findAllByProps({ className: 'w-100 py-5 text-center' })
        const jsonTitle = compTitle.toJSON()
        expect(jsonTitle).toMatchSnapshot()
        expect(resultRoot).toHaveLength(1)
    })
    test('Sould be edit insted register', () => {
        const compTitle = renderer.create(<Title idserver={22} />)
        const resultRoot = compTitle.root.findAllByProps({ ['data-test']: 'Edit the Product - 22' })
        expect(resultRoot).toHaveLength(1)
    })
    test('Should not render without the schema', () => {
        schemaMocked.default = null
        const compTitle = renderer.create(<Title idserver={12} />)
        const result = compTitle.toJSON()
        expect(result).toBeNull()
    })
})