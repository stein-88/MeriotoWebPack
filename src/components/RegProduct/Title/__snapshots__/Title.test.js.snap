// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Title Component Should render 1`] = `
<section
  className="w-100 py-5 text-center"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6"
      >
        <section
          className="w-100"
        >
          <p
            className="h1"
            data-test="Register the Product"
          >
            Register the Product
          </p>
        </section>
      </div>
      <div
        className="col-6"
      >
        <section
          className="w-100"
        >
          <p
            className="h1"
          >
            See how will be
          </p>
        </section>
      </div>
    </div>
  </div>
</section>
`;
